src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs

```diff
@@ -103,7 +103,19 @@ protected override async Task<ExitCode> InvokeInternal(ILogger logger)
                 token.WaitHandle.WaitOne();
             }
 
-            driver.Quit();  // Firefox driver hangs if Quit is not issued.
+            // close all tabs before quit is a workaround for broken Selenium - GeckoDriver communication in Firefox
+            // https://github.com/dotnet/runtime/issues/101617
+            logger.LogInformation($"Closing {driver.WindowHandles.Count} browser tabs before setting the main tab to config page and quitting.");
+            while (driver.WindowHandles.Count > 1)
```

Comment:
nit: it'd be nice to have some (short) timeout here so this doesn't end up being an infinite loop if closing a tab doesn't work for some reason